haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,368 @@
+        if not model_apapter_cls:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not supported by this invocation layer."
+            )
+        self.model_adapter = model_apapter_cls(model_kwargs=model_input_kwargs, max_length=self.max_length)
+
+    def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union[str, List[Dict[str, str]]]:
+        # the prompt for this model will be of the type str
+        if isinstance(prompt, List):
+            raise ValueError("SageMaker invocation layer doesn't support a dictionary as prompt, only a string.")
+
+        resize_info = self.prompt_handler(prompt)
+        if resize_info["prompt_length"] != resize_info["new_prompt_length"]:
+            logger.warning(
+                "The prompt has been truncated from %s tokens to %s tokens so that the prompt length and "
+                "answer length (%s tokens) fit within the max token limit (%s tokens). "
+                "Shorten the prompt to prevent it from being cut off.",
+                resize_info["prompt_length"],
+                max(0, resize_info["model_max_length"] - resize_info["max_length"]),  # type: ignore
+                resize_info["max_length"],
+                resize_info["model_max_length"],
+            )
+        return str(resize_info["resized_prompt"])
+
+    @classmethod
+    def supports(cls, model_name_or_path, **kwargs):
+        model_supported = cls.get_model_adapter(model_name_or_path) is not None
+        if not model_supported or not cls.aws_configured(**kwargs):
+            return False
+
+        try:
+            session = cls.get_aws_session(**kwargs)
+            bedrock = session.client("bedrock")
+            foundation_models_response = bedrock.list_foundation_models(byOutputModality="TEXT")
+            available_model_ids = [entry["modelId"] for entry in foundation_models_response.get("modelSummaries", [])]
+            model_ids_supporting_streaming = [
+                entry["modelId"]
+                for entry in foundation_models_response.get("modelSummaries", [])
+                if entry.get("responseStreamingSupported", False)
+            ]
+        except AWSConfigurationError as exception:
+            raise AmazonBedrockConfigurationError(message=exception.message) from exception
+        except Exception as exception:
+            raise AmazonBedrockConfigurationError(
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly."
+            ) from exception
+
+        model_available = model_name_or_path in available_model_ids
+        if not model_available:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not available in Amazon Bedrock. "
+                f"Please make sure the model is available in the configured AWS region and you've been granted access."
```

Comment:
```suggestion
                f"Make sure the model you want to use is available in the configured AWS region and you have access."
```
where should have they access? to the model, to the region?
should we tell them more about how to check if a model is available?

Reply:
No, anybody who has access to a bedrock model will know what to do.